bin/cheribsdtest/cheribsdtest_syscall.c

```diff
@@ -62,30 +62,6 @@
 
 #include "cheribsdtest.h"
 
-CHERIBSDTEST(sig_dfl_neq_ign, "Test SIG_DFL != SIG_IGN")
-{
-	void * __capability sic = (__cheri_tocap void * __capability)SIG_IGN;
-
-	/*
-	 * This may appear redundant, but it tickles some of the optimizers
-	 * differently than the condition below, which, apparently, can get
-	 * constant-folded and DCE'd, while this does not.
-	 */
-	int eq = (SIG_IGN == SIG_DFL);
```

Comment:
This is checking that SIG_IGN is defined in a way that doesn't use CToPtr/CFromPtr, so it is more of an OS headers test than a compiler test. git blame points at https://github.com/CTSRD-CHERI/cheribsd/commit/1e704e0f3837a5de493598449408de62ff6a8db6

Reply:
The talk of constant-folding is a bit compiler-facing, though, and fragile. Probably if we really want to make sure we are careful we should just be using volatile variables to capture the actual value of each constant.

Reply:
I agree this test is rather useless now that the underlying issue has been fixed. However, I do wonder if we would find the same issue if we were to port cheribsdtest to e.g. linux.

Reply:
https://git.morello-project.org/morello/kernel/linux/-/blob/morello/master/include/uapi/asm-generic/signal-defs.h#L88-90 🙃

Reply:
Could we safely keep a version that is just the tail end of the test that requires the constants to test !=?

Reply:
That should be fine, it's able to detect the buggy case from before https://cheri-compiler-explorer.cl.cam.ac.uk/z/EMnYEW